### PR TITLE
Feature/ui transmission interval

### PR DESCRIPTION
**Pull request by dtrabo**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Add UI to Sasquatch app to set transmission interval when it sends event logs.
Screenshot 2019-05-16 at 6 27 20 PM

## Related PRs or issues

[AB#59982](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/59982)

### CODE REVIEW

File: Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift [hunk 1]
```diff
@@ -241,16 +246,34 @@ class MSAnalyticsViewController: UITableViewController, AppCenterProtocol {
...
+  
+  @objc func doneClicked() {
+    updateIntervalLabel(transmissionInterval: Int(transmissionIntervalField.text!))
+    UserDefaults.standard.setValue(Int(transmissionIntervalField.text!), forKey: kMSTransmissionIterval)
+    dismissKeyboard(self.transmissionIntervalField)
+  }
+  
+  func updateIntervalLabel(transmissionInterval: Int?) {
+    let (h, m, s) = secondsToHoursMinutesSeconds(seconds: transmissionInterval ?? 3)
+    let convertInterval = String(format: "%02d:%02d:%02d",h ,m ,s)
```
Comment (russelarms) [inline Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift]: is it properly formatted?

Comment (annakocheshkova) [inline Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift]: Teeny tiny nitpick
```suggestion
    let convertInterval = String(format: "%02d:%02d:%02d", h, m, s)
```

File: Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift [hunk 2]
```diff
@@ -241,16 +246,34 @@ class MSAnalyticsViewController: UITableViewController, AppCenterProtocol {
...
+    transmissionIntervalLabel.text = "sec " + "(" + convertInterval + ")"
```
Comment (annakocheshkova) [inline Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift]: Combine this with previous String.format.

Comment (russelarms) [inline Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift]: It's better to use escape operator and one pair of quotes.

File: Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift [hunk 3]
```diff
@@ -241,16 +246,34 @@ class MSAnalyticsViewController: UITableViewController, AppCenterProtocol {
...
+    return toolbar
+  }
+  
+  @objc func doneClicked() {
+    updateIntervalLabel(transmissionInterval: Int(transmissionIntervalField.text!))
+    UserDefaults.standard.setValue(Int(transmissionIntervalField.text!), forKey: kMSTransmissionIterval)
+    dismissKeyboard(self.transmissionIntervalField)
+  }
+  
+  func updateIntervalLabel(transmissionInterval: Int?) {
```
Comment (annakocheshkova) [inline Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift]: Is it really Optional?

**Comment by annakocheshkova** [comment]
@MatkovIvan it will overlap. We need a shorter name then

File: Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift [hunk 4]
```diff
@@ -241,16 +246,34 @@ class MSAnalyticsViewController: UITableViewController, AppCenterProtocol {
...
-    self.latencyField.delegate = self.latencyPicker
-    self.latencyField.text = self.latency.rawValue
-    self.latencyField.tintColor = UIColor.clear
+    transmissionIntervalField.inputAccessoryView = self.toolBarForKeyboard()
+    let interval = UserDefaults.standard.integer(forKey: kMSTransmissionIterval) != 0 ? UserDefaults.standard.integer(forKey: kMSTransmissionIterval) : 3
+    transmissionIntervalField.text = String(interval)
+    updateIntervalLabel(transmissionInterval: interval)
+  }
+  
+  func secondsToHoursMinutesSeconds(seconds : Int) -> (Int, Int, Int) {
```
Comment (Jamminroot) [inline Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift]: I'd go for built-in functionality. [DateComponentsFormatter](https://developer.apple.com/documentation/foundation/datecomponentsformatter)

File: Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift (hunk 1 above)
Comment (Jamminroot) [inline Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift]: Same here - we can just use DateComponentsFormatter

File: Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift (hunk 2 above)
Comment (dtrabo) [reply Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift]: No longer relevant

Comment (dtrabo) [reply Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift]: No longer relevant

File: Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift (hunk 3 above)
Comment (dtrabo) [reply Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift]: Rewrote, now it's not Optional

File: Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift (hunk 1 above)
Comment (dtrabo) [reply Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift]: Rewrote with DateComponentsFormatter

File: Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift [hunk 5]
```diff
@@ -200,6 +200,8 @@ class MSAnalyticsViewController: UITableViewController, AppCenterProtocol {
     tableView.deselectRow(at: indexPath, animated: true)
     if indexPath.section == kEventPropertiesSectionIndex && eventPropertiesSection.isInsertRow(indexPath) {
       self.tableView(tableView, commit: .insert, forRowAt: indexPath)
+    } else if (indexPath.section == 0 && indexPath.row == 3){
```
Comment (MatkovIvan) [inline Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift]:
```suggestion
    } else if indexPath.section == 0 && indexPath.row == 3 {
```

File: Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift [hunk 6]
```diff
@@ -240,17 +242,44 @@ class MSAnalyticsViewController: UITableViewController, AppCenterProtocol {
...
+    formatter.zeroFormattingBehavior = [ .pad]
+    transmissionIntervalLabel.text = formatter.string(from: formattedInterval)
+  }
+  
+  func initTransmissionAlert(_ tableView: UITableView) -> UIAlertController {
+    let alert = UIAlertController(title: "Transmission Interval", message: nil, preferredStyle: .alert)
+    let confirmAction = UIAlertAction(title: "OK", style: .default, handler: {(_ action:UIAlertAction) -> Void in
+      let result = alert.textFields?[0].text
+      var timeResult: Int = Int(result!) ?? 3
+        if timeResult > 86400 {
```
Comment (MatkovIvan) [inline Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift]: Please allow entering invalid values to verify the SDK validation logic

File: Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift [hunk 7]
```diff
@@ -240,17 +242,44 @@ class MSAnalyticsViewController: UITableViewController, AppCenterProtocol {
...
-  func initLatencyPicker() {
-    let latencyPosition = Latency.allTimeValues.index(of: UserDefaults.standard.integer(forKey: kMSTransmissionIterval))
-    self.latencyPicker = MSEnumPicker<Latency>(
-    textField: latencyField,
-    allValues: Latency.allValues,
-    onChange: { index in
-      UserDefaults.standard.setValue(Latency.allTimeValues[index], forKey: kMSTransmissionIterval)
+  func initTransmissionIntervalLabel() {
+    if UserDefaults.standard.integer(forKey: kMSTransmissionIterval) == 0 {
+      UserDefaults.standard.setValue(3, forKey: kMSTransmissionIterval)
```
Comment (MatkovIvan) [inline Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift]: Rely on SDK value, do not save the default value here

**Pull request merged**